pkg/crc/preflight/preflight_checks_common.go

```diff
@@ -254,3 +266,33 @@ func fixGVProxyExecutableCached() error {
 	}
 	return nil
 }
+
+// Check if macadam executable is cached or not
+func checkMacadamExecutableCached() error {
+	if version.IsInstaller() {
+		return nil
+	}
+
+	macadam := cache.NewMacadamCache()
+	if !macadam.IsCached() {
+		return errors.New("macadam executable is not cached")
+	}
+	if err := macadam.CheckVersion(); err != nil {
+		return errors.Wrap(err, "unexpected version of the macadam executable")
```

Comment:
Let’s not add new uses of `errors.Wrap` in new code.
https://github.com/pkg/errors
>  This repository was archived by the owner on Dec 1, 2021. It is now read-only.